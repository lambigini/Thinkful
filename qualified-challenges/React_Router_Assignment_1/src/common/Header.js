import React from "react";

export const Header = () => (
  <div className="jumbotron jumbotron-fluid bg-warning">
    <div className="container">
      <h1 className="display-4">Community Blog</h1>
      <p className="lead">Check out the posts of our community members.</p>
    </div>
  </div>
);

export default Header;
