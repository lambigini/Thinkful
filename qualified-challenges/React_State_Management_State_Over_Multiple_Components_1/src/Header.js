import React from "react";

function Header({ loggedIn, handleLoggedInClick }) {
  return (
    <button onClick={handleLoggedInClick}>
      {loggedIn ? "Log Out" : "Log In"}
    </button>
  );
}

export default Header;
