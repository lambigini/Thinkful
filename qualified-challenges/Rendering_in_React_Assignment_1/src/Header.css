.header {
  display: flex;
}
div div {
  flex: 1;
}
img {
  width: 100%;
}

.info {
  text-align: center;
}
