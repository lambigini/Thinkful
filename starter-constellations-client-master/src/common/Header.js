import React from "react";
import Search from "../home/Search";

function Header({ constellations, setConstellations }) {
  return (
    <div className="jumbotron jumbotron-fluid bg-info text-white rounded-0 mb-0">
      <div className="container">
        <h1 className="display-4">I love Constellations</h1>
        <p className="lead">Explore different constellations in the sky!</p>
        <hr />
        <Search
          constellations={constellations}
          setConstellations={setConstellations}
        />
      </div>
    </div>
  );
}

export default Header;
