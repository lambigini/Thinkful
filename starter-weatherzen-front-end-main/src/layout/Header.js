import React from "react";
import headerImage from "./header.jpg";

const style = {
  background: `linear-gradient(rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.7)), url(${headerImage})`,
  backgroundPosition: "center",
  backgroundSize: "100% auto",
};

function Header() {
  return (
    <div className="jumbotron jumbotron-fluid text-white mb-0" style={style}>
      <div className="container">
        <h1 className="display-1">WeatherZen</h1>
        <p className="lead">Crowdsourcing local weather observations!</p>
      </div>
    </div>
  );
}

export default Header;
