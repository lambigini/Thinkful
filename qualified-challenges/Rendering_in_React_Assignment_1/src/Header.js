import React from "react";
import "./Header.css";

function Header({ name, imageSrc, birthday }) {
  return (
    <div className="header">
      <div className="img">
        <img src={imageSrc} alt="white-cat" />
      </div>

      <div className="info">
        <h1>{name}</h1>
        <h2>{birthday}</h2>
      </div>
    </div>
  );
}

export default Header;
