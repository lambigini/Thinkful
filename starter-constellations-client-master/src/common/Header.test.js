import React from "react";
import Header from "./Header";
import renderer from "react-test-renderer";

it("renders the header with a search bar", () => {
  const constellations = [
    {
      id: "zb8QvVt",
      name: "Crater",
      meaning: "Cup",
      starsWithPlanets: 10,
      quadrant: "SQ2",
    },
  ];
  const component = renderer.create(<Header constellations={constellations} />);
  expect(component.toJSON()).toMatchSnapshot();
});
