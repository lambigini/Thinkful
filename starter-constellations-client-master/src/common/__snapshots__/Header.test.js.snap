// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the header with a search bar 1`] = `
<div
  className="jumbotron jumbotron-fluid bg-info text-white rounded-0 mb-0"
>
  <div
    className="container"
  >
    <h1
      className="display-4"
    >
      I love Constellations
    </h1>
    <p
      className="lead"
    >
      Explore different constellations in the sky!
    </p>
    <hr />
    <form
      className="form container"
    >
      <div
        className="form-group row"
      >
        <input
          className="form-control col col-md-8"
          id="search"
          name="search"
          onChange={[Function]}
          placeholder="Search the stars..."
          type="text"
        />
        <label
          className="form-check-label sr-only"
          htmlFor="search"
        >
          Search
        </label>
      </div>
    </form>
  </div>
</div>
`;
